platform-core-module/src/main/java/com/softicar/platform/core/module/transactional/data/TransactionalDataDeleter.java

```diff
@@ -0,0 +1,83 @@
+package com.softicar.platform.core.module.transactional.data;
+
+import com.softicar.platform.common.core.exceptions.SofticarUserException;
+import com.softicar.platform.core.module.CoreI18n;
+import com.softicar.platform.core.module.module.IModule;
+import com.softicar.platform.db.core.transaction.DbTransaction;
+import com.softicar.platform.db.runtime.table.IDbTable;
+import com.softicar.platform.db.runtime.table.dependency.DbTableDependencyGraph;
+import com.softicar.platform.emf.module.registry.CurrentEmfModuleRegistry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class TransactionalDataDeleter {
```

Comment:
As this class poses a significant threat to the DB content (by design), we should have tests to ensure that it deletes _exactly_ what it's supposed to.

Reply:
- hmm, can hardly be tested without payload modules...any suggestions?
- changed popup to display the statements that will be executed, IMO, it's the obligation of the admin to verify those
![image](https://user-images.githubusercontent.com/83839745/199996447-857992c8-0c01-4cf8-938f-bb07718260c7.png)

Reply:
implemented test in code module